src/main/java/Reverser.java

```diff
@@ -0,0 +1,45 @@
+import java.util.StringJoiner;
+
+public class Reverser {
+
+    public static String reverseOnlyWords (String str){
+        String[] words = checkNull(str).split(" ");
+        return reverseOneWord(words);
+    }
+
+    private static String checkNull (String str) {
+        String checkvalue = str;
+        if  (checkvalue == null){
+            throw new IllegalArgumentException("The string value must not be null");
+        }
+        return checkvalue;
+    }
```

Comment:
Method can be void, because you don't do anything with String str. In this method can be only if-block, without "String checkvalue = str" and "return checkvalue"